### DISCUSSION TITLE
supabase email password authentication in Flutter returning null user

### DISCUSSION THREAD

**Discussion by sumchans** [supabase/supabase#1901] [Questions] [answered]
Here is my code to signup a user using email password in Flutter, the user gets created which I can see in supabase users list, but the response contains null user value .
```
Future<void> signUp() async {
  final response = await supabaseClient.auth.signUp(
    'test@test.com',
    'testpass',
  );
  print(response.user);
}
```

**Comment by phamhieu** [comment, marked as answer]
Because the user needs to confirm the email address first. On signUp, if:
- email verification IS required: no session/user return
- email verification NOT required: session/user return

If you don't need email verification, you can turn it off in your Supabase project Authentication settings.

<img width="386" alt="Screenshot 2021-07-06 at 2 08 43 PM" src="https://user-images.githubusercontent.com/689843/124557569-da44c280-de63-11eb-9ff0-dc9a8ad30102.png">
